include/mxnet/c_api.h

```diff
@@ -643,14 +643,14 @@ MXNET_DLL int MXNDArrayCreateSparseEx(int storage_type,
 
 MXNET_DLL int MXNDArrayCreateSparseEx64(int storage_type,
                                         const int64_t *shape,
-                                        int ndim,
+                                        uint32_t ndim,
```

Comment:
Isn't ndim type always int? why change it here?

Reply:
Number of dimensions in Large tensor would still be <2**32 right?
Value of single dimension could be > 2**32.

Reply:
> Isn't ndim type always int? why change it here?

keeping it consistent with 32bit version: https://github.com/apache/incubator-mxnet/pull/16898/files#diff-0ce80a2c739b88d1b42356fc2aff2630L633

Reply:
Why not make everything to int to be consistent? I think internally ndim is always int.

Reply:
@apeforest just to clarify you want to change 32bit C API signature as well ?
If we do that then it might be a behaviour change for this particualt C API
If we only change it for 64 bit. Then https://github.com/apache/incubator-mxnet/pull/16898/files#diff-3288207540b433242fe8da1390c68d3fR852 
method would require additional template parameter to differentiate ndim type to be used for 32bit version and 64 bit version.

Reply:
i c. In that case, I guess we can use uint32_t for both 32-bit and 64-bit API. But I would also check with Numpy project owner to see if -1 is not used for numpy operator. Thanks!

Reply:
@reminisce Confirmed with him. -1 is used to indicate "unknown". Internally mxnet does use int. Hence it makes sense to add 64bit API with ndim as int. 
Since 32 bit API existed before with uint32_t, not sure if it makes sense to have a API breaking change.

Reply:
Thanks for the discussion. Using `int` for `ndim` is consistent with MXNet convention.